src/stdlib_math.fypp

```diff
@@ -0,0 +1,35 @@
+#:include "common.fypp"
+
+#:set INT_KINDS_TYPES = [("int8", "integer"), ("int16", "integer"), ("int32", "integer"), ("int64", "integer")]
+#:set REAL_KINDS_TYPES = [("sp", "real"), ("dp", "real"), ("qp", "real")]
+#:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+
+module stdlib_math
+    use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
+
+    implicit none
+    private
+    public :: clip
+
+
+    interface clip
+    #:for k1, t1 in IR_KINDS_TYPES
+        module procedure clip_${t1}$_${k1}$
+    #:endfor
+    end interface clip
+
+
+    contains
```

Comment:
A bit odd indentation, this way you will end up using all your line space quickly, especially with the 4 space indentation. I'd recommend to unindent the following block by 4 spaces.